perllib/Open311/PopulateServiceList.pm

```diff
@@ -283,6 +285,21 @@ sub _set_contact_group {
     }
 }
 
+sub _set_contact_non_public {
+    my ($self, $contact) = @_;
+
+    # We never want to make a private category unprivate.
+    return if $contact->non_public;
+
+    my $keywords = $self->_current_service->{keywords} || '';
```

Comment:
For future proofing, do we want to do something like `%keywords = map { $_ => 1 } split /,/, $keywords` and use `$keywords{private}` just in case a keyword called "notatallprivate" came along? Unlikely I realise!